### DISCUSSION TITLE
Development Roadmap

### DISCUSSION THREAD

**Discussion by rhahao** [sws2apps/organized-app#178] [Announcements]
Discover the features already available in Organized and see what’s in development or coming soon. Our goal is to help you save time on your responsibilities while still achieving better scheduling results. Take a look, and feel free to share it with others who might be interested in Organized’s development progress!

### Meetings
  - [x] Meeting materials auto-import (from .epub/jw.org)
      - [x] Separate languages for UI and meeting materials
  - [x] Midweek meeting scheduling
      - [x] Midweek meeting scheduling – full month view
  - [x] Weekend meeting scheduling
  - [x] Autofill for meeting assignments
  - [x] Public talks – local and outgoing
  - [x] Visiting and local speakers catalog
  - [x] Print schedules templates
  
### Congregation
  - [x] Field service groups
    - [x] Service groups PDF export
    - [x] Assign a group upon person creation
  - [x] Import/export of the app data
  - [x] Foreign language groups
  - [x] Upcoming events

  ### Persons
  - [x] Assignment qualifications
  - [x] Privileges and enrollments
  - [x] Assignment history
  - [x] Search field and filters
  - [x] Add person shortcut
  - [x] Quick shortcut to creating a user 
  - [x] Time away
  - [x] Emergency contacts
  - [x] Import Persons with CSV
  ### Reports 
  - [x] Publisher records
  - [x] Field service report
    - [x] Reports of delegated persons
  - [x] Meeting attendance record
    - [x] Export S-88
  - [x] Congregation’s field service and meeting attendance (S-1)
  
  ### Ministry 
  - [x] Congregation analysis report (S-10)
  - [x] Ministry timer
  - [x] Auxiliary pioneer application
  
  ### Security
  - [x] Two-factor authentication
  - [x] End-to-end encryption
  - [x] Two-level data encryption
  - [x] Congregation access code
  - [x] User rights and role management
  - [x] User sessions control
  - [x] Family members

  ### User account
  - [x] Time away
  - [x] View and terminate user sessions 
  - [x] Color schemes
  - [x] Dark and light mode
    - [x] Automatic mode change   
  - [x] Enable deleting user account
  
  ### General
  - [x] Keyboard navigation for improved accessibility
  
  ### Organized ecosystem 
  - [x] Test mode
  - [x] [About Organized](https://about.organized-app.com/) website
  - [x] Detailed [User guide](https://guide.organized-app.com)
  - [x] Video tutorials on our [YouTube channel](https://www.youtube.com/@organized-app/featured)
    - [x] Meet Organized
    - [x] Register as baptized brother
    - [x] Register as publisher
    - [x] Why we build Organized
    - [x] Register your congregation
    - [x] How to translate
    - [ ] Invite users
    - [ ] Add persons

---

## 🟡 Actively developed
- [x] Field service meetings
- [x] Meeting duties
- [x] App PIN lock
- [x] Settings page redesign
- [x] Information board
- [x] Public witnessing
- [x] Auditorium attendant mode
- [ ] Self-hosted version of the app

---

### Meetings
- [ ] Hall cleaning

### Ministry
- [ ] Territories
- [ ] Campaigns
- [ ] Find a preaching partner

### Congregation
- [ ] Collaborative activities
- [ ] Literature management
- [ ] Import data from other apps

---

## ⚪️ On our radar
- [ ] LDC maintenance
- [ ] Elder meetings scheduling
- [ ] Meeting chairman mode
- [ ] Circuit overseer collaboration
- [ ] Duties distribution overview
- [ ] Congregation persons overview
- [ ] Push notifications
- [ ] Advanced territories features

If you have questions, please do not hesitate to post your comment.

Warm regards! 😉

**Comment by luannsct** [comment]
Hello, I thought the project was wonderful, I would like to help with my knowledge. How can we do it?

**Reply by rhahao** [reply]
Hello @luannsct: thanks for your interest in helping us developing this app. Just to let you know that we are now in the process of building a new redesigned version of this app (I’ve sent you a DM with additional details related to that. 😁) No new features will be added to the current version, but PR is always welcomed to fix bugs or address some issues. [Contributing docs](https://github.com/sws2apps/organized-app/blob/main/CONTRIBUTING.md)

**Reply by romeoagyemang** [reply]
I am also interested, I want to help

**Reply by FranklinD10** [reply]
Me too. I am interested and want to help. I have been assisting with bugs, but I am happy to assist further.

**Reply by ux-git** [reply]
@FranklinD10 thanks for your first contribution. Can't wait for it to be reviewed and merged 🙂

**Comment by Advoit** [comment]
Hello, first of all, I would like to thank you for this great project. It looks very good already. I would like to know if there is already an approximate release date for the self-hosted version of the app.

**Reply by ux-git** [reply]
Thank you @Advoit

The self-hosted version will require quite a lot of changes in the code, so we set as a goal to first develop all the essential features for the scheduling and other key core features before we dive deep into the self-hosted version.

So there are no approximate estimates. It will likely take time.

**Comment by yajo** [comment]
May I ask progress about the ongoing developments?

I see #4369 and #4210 are stale since months ago, and are precisely the two features I'm missing the most.

Thank you.

**Reply by ux-git** [reply]
Thanks for checking in 😊
Both of those are still in draft status at the moment. They’re planned, but development hasn’t actively resumed on them yet.

At the moment, we’re focusing on improving the sync structure and logic to make the overall data sync experience better. Once that’s in place, we’ll resume working on those features.

**Comment by hrasamoe** [comment]
Hello, thank you for this great project. I’m very interested in the upcoming features, especially meeting ministry and meeting duties. Could you share if there are already mockups or design plans for these modules? I’d be glad to give feedback or contribute.

Thank you

**Reply by ux-git** [reply]
Hey @heri1223samuel, thanks for your interest in the app! We do already have the designs of those two features and actually we do have two draft PRs for those that need to be worked on and finalized. If you wish to contribute with the development of the [ministry meetings](https://github.com/sws2apps/organized-app/pull/4369) feature, feel free to contact us at support@organized-app.com to request an access and then discuss how we can collaborate. The UI for the meetings is almost done, it needs to be finalized and connected to the database and backend.

**Reply by hrasamoe** [reply]
Hello, I contacted support team two days ago and received the link, but I still don't have access to the Figma project. Could you please check this? Thank you!

**Reply by ux-git** [reply]
@heri1223samuel Done 👍

**Comment by SecDD** [comment]
Thanks for having this on the roadmap — it’s really appreciated.

Do you have a rough idea of the timeline for the announcements / information board feature? From a congregation perspective, this is one of the key missing pieces at the moment.

Also, would there be a possibility to include a few basic additions such as:

- Attachments (correspondence for the cong.)
- optional read/seen indication for important announcements

Even a simple version of this feature would already make a big difference in reducing reliance on external tools.

Thanks again for the work being done.

**Reply by ux-git** [reply]
Hey @SecDD! Thanks a lot, we really appreciate it. The information board feature hasn’t been a top priority so far, so we have no approximate timeline yet. Thanks to your message we will keep an eye on it to work on it as soon as we have finished other pending tasks. The design is actually already done, it’s just waiting its turn.

For attachments, we need to figure out a sustainable approach, since we try to keep as little data stored online as possible. So that part needs more thought before we implement it properly.

Could you share a bit more about how you imagine the read/seen feature working? Do you mean something like a notification sent to all publishers with a “Mark as read” option, or more of a read/seen status directly on the information board?

**Reply by SecDD** [reply]
Thanks so much for the response. Very encouraging to hear the design is already done.

Regarding the read/seen feature, I was thinking more from a congregation usability perspective than as a messaging-style feature.

Rather than “mark as read” notifications requiring action, I was imagining a simple optional read/seen indicator directly on the information board for announcements marked as important.

The value would be practical rather than administrative, for example helping elders know whether important information has likely reached the congregation (special arrangements, schedule changes, urgent reminders, etc.), without needing separate follow-ups through other tools.

Something lightweight could already help, for example:

- optional “seen by X publishers” status on an announcement
- or an optional per-announcement viewed indicator for important notices

I’d see it more as confirmation that information has been noticed, not as tracking or requiring responses.

And regarding attachments, your point makes sense. Even support for simple links in announcements could already cover many use cases while keeping storage minimal.

Thanks again for considering this. I really appreciate the thought going into it.

**Reply by ux-git** [reply]
Thanks a lot for the additional information. I’ve taken notes, and we’ll see if implementing the "Seen" indicator is feasible. Meanwhile, we’ve reviewed image and PDF compression solutions to minimize file sizes on user devices before uploading. There are some promising options to reduce storage usage, so we're excited to see this feature come to life.

**Reply by Rickyb1194** [reply]
I think it would be great if annoucnements could be specific per group - for example: elders meeting.  only the elders see this on their board and are notified of it.  or pioneer meeting.  same thing.

**Comment by Rickyb1194** [comment]
Hello, I love what you are doing and would love to help with my time.  I don't know how to code, but have built apps with claude code (i know, its not coding) and they work great.  not sure if building a branch of a feature you guys want to add and letting you guys test it is of value or not, but if it is, I would love to help you get some more of these features in the app.

**Reply by ux-git** [reply]
Hey @Rickyb1194, thanks for your message. 


If you have any ideas, a good starting point might be suggesting some smaller UI improvements to get a feel for how everything works. @rhahao will review the code anyway to ensure the changes are suitable and function well.

If you'd like to discuss specific features before working on them, just dive deeper into the app, see what can be fixed or improved, and drop us a few suggestions as an issue. We'll take a look and let you know what is worth implementing.

How does that sound to you?
